example/src/main/java/org/aerogear/mobile/example/ui/AuthFragment.java

```diff
@@ -41,23 +42,32 @@ public void doLogin() {
         AuthService authService = ((MainActivity) getActivity()).getAuthService();
         DefaultAuthenticateOptions authOptions =
                         new DefaultAuthenticateOptions(this.getActivity(), LOGIN_RESULT_CODE);
-        try {
-            authService.login(authOptions, new Callback<UserPrincipal>() {
-                @Override
-                public void onSuccess(UserPrincipal models) {
-                    // user logged in, continue on..
-                    Log.i(TAG, "user logged in " + models.toString());
-                    ((MainActivity) getActivity()).navigateToAuthDetailsView(models);
-                }
-
-                @Override
-                public void onError(Throwable error) {
-                    // there is an error during the login
-                    Log.e(TAG, "login failed due to error " + error.getLocalizedMessage());
-                }
-            });
-        } catch (IllegalStateException e) {
-            Log.e("Pinning Error", e.getMessage());
-        }
+        authService.login(authOptions, new Callback<UserPrincipal>() {
+            @Override
+            public void onSuccess(UserPrincipal models) {
+                // user logged in, continue on..
+                Log.i(TAG, "user logged in " + models.toString());
+                ((MainActivity) getActivity()).navigateToAuthDetailsView(models);
+            }
+
+            @Override
+            public void onError(Throwable error) {
+                // there is an error during the login
+                Log.e(TAG, "login failed due to error " + error.getLocalizedMessage());
+                failureDialog(error.getLocalizedMessage());
+            }
+        });
+    }
+
+    public void failureDialog(String e) {
+        String error = e;
```

Comment:
What is the use of this?

Reply:
Good catch, left over from a different approach I was taking